packages/client/hmi-client/src/components/model-template/tera-model-template.vue

```diff
@@ -1,150 +1,163 @@
 
-function updateNewCardPosition(event) {
-	newCard.x = (event.offsetX - canvasTransform.x) / canvasTransform.k;
-	newCard.y = (event.offsetY - canvasTransform.y) / canvasTransform.k;
-}
+const fakeVariables = ['X', 'Y', 'p'];
 
-function onDrop(event) {
-	updateNewCardPosition(event);
-	cards.value.push(cloneDeep(newCard));
+function updateName() {
+	emit('update-name', name);
+	isEditingName.value = false;
 }
-
-const updatePosition = (card: ModelTemplateCard, { x, y }) => {
-	if (!isMouseOverCanvas) return;
-	card.x += x / canvasTransform.k;
-	card.y += y / canvasTransform.k;
-};
 </script>
 
 <style scoped>
-aside {
-	width: 15rem;
+.card-container {
 	display: flex;
-	flex-direction: column;
-	height: 100%;
-	background-color: #f4f7fa;
-	border-right: 1px solid var(--surface-border-alt);
-	padding: 1rem;
-	gap: 0.5rem;
+	font-size: var(--font-caption);
+}
+
+.card {
+	display: flex;
+	background-color: var(--surface-section);
+	border-radius: var(--border-radius-medium);
+	outline: 1px solid var(--surface-border-alt);
+	min-width: 12rem;
+	position: relative;
+	box-shadow:
+		0px 1px 3px 0px rgba(0, 0, 0, 0.08),
+		0px 1px 2px 0px rgba(0, 0, 0, 0.04);
+
+	& > main {
+		width: 100%;
+		margin: 0.5rem;
+		display: flex;
+		flex-direction: column;
+		gap: 1rem;
+
+		& > header {
+			cursor: pointer;
+			text-align: center;
+			height: 2rem;
+
+			& > .p-inputtext.p-inputtext-sm {
+				padding: 0.2rem 0.3rem;
+				text-align: center;
+			}
+		}
+
+		& > * {
+			margin: 0 auto;
+		}
+	}
+
+	& > .p-button {
+		display: none;
+		position: absolute;
+		bottom: 0;
+		right: 0;
+	}
 }
 
 ul {
 	list-style: none;
 	display: flex;
 	flex-direction: column;
 	gap: 0.5rem;
-	margin-top: 0.5rem;
+	margin: 0.5rem 0;
+
+	& > li {
+		background-color: var(--surface-section);
+		border: 1px solid var(--surface-border-alt);
+		border-top-right-radius: var(--border-radius);
+		border-bottom-right-radius: var(--border-radius);
+		padding: 0.15rem 0.25rem;
+		color: var(--text-color-subdued);
+		box-shadow:
+			0px 1px 3px 0px rgba(0, 0, 0, 0.08),
+			0px 1px 2px 0px rgba(0, 0, 0, 0.04);
+		/* Font should be "Latin Modern Math" */
+		font-family: serif;
+		font-style: italic;
+	}
 }
 
-h5 {
+.draggable {
+	width: 0.75rem;
+	border-top-left-radius: var(--border-radius-medium);
+	border-bottom-left-radius: var(--border-radius-medium);
+	background-color: var(--surface-highlight);
 	display: flex;
 	align-items: center;
-	gap: 0.25rem;
-	font-weight: var(--font-weight);
-}
 
-header {
-	color: var(--text-color-subdued);
-	font-size: var(--font-caption);
+	& > .pi {
+		font-size: 0.75rem;
+		color: var(--text-color-subdued);
+	}
 }
+</style>
 
-.pi-info-circle {
-	color: var(--text-color-subdued);
-	cursor: help;
-}
+<style>
+/* When a card is placed in the data layer of the infinite canvas 
+(eg. ports shouldn't look selectable if card is in the sidebar) */
+.data-layer .card-container {
+	& .card:hover > .p-button {
+		display: block;
+	}
```

Comment:
Not a fan of this kind of coupling across components, as it is prone to creating bugs. 
We can leave this in as-is for this PR, but I think we should think of ways to have the constraint in a more spelled-out manner, maybe something like:

```
<tera-model-template :readonly="true" /> 
```

or something to this effect.